src/App.tsx

```diff
@@ -1,14 +1,49 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilterOptions } from './types/FilterOptions';
 
 export const App: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [query, setQuery] = useState('');
+  const [filterOption, setFilterOption] = useState<FilterOptions>('all');
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(tasks =>
+        tasks.filter(task => {
+          switch (filterOption) {
+            case 'all':
+              return true;
+
+            case 'active':
+              return !task.completed;
+
+            case 'completed':
+              return task.completed;
+          }
+        }),
+      )
+      .then(tasks =>
+        tasks.filter(task =>
+          task.title.toLowerCase().includes(query.toLowerCase()),
+        ),
+      )
+      .then(setTodos)
+      .finally(() => setIsLoading(false));
```

Comment:
looks too complex, you will make every time on query or filterOption change new request to BE

just filter the array like that in the component
`const filteredTodos = getFilteredTodos(todos, query, filterOption)`
don't forget to create that function